pandas/core/generic.py

```diff
@@ -7680,6 +7679,38 @@ def between_time(
         if not isinstance(index, DatetimeIndex):
             raise TypeError("Index must be DatetimeIndex")
 
+        # if any of the depreciated arguments ('include_start', 'include_end')
+        # have been passed
+        if (include_start != lib.no_default) or (include_end != lib.no_default):
+            warnings.warn(
+                "`include_start` and `include_end` are deprecated in"
+                "favour of `inclusive`.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            if include_start == lib.no_default:
+                include_start = True
+            if include_end == lib.no_default:
+                include_end = True
+        else:  # if depreciated args haven't been passed
+            if inclusive == "both":
+                include_start = True
+                include_end = True
+            elif inclusive == "neither":
+                include_start = False
+                include_end = False
+            elif inclusive == "left":
+                include_start = True
+                include_end = False
+            elif inclusive == "right":
+                include_start = False
+                include_end = True
```

Comment:
if you also changed `DateTimeIndex.indexer_between_time` then this would not be necessary

Reply:
actually see changes below for another simplification.

Reply:
Should I change `DateTimeIndex.indexer_between_time`? I see that it is being used in other files as well.

Reply:
@attack68 please take a look :)

Reply:
I'm modifying `DateTimeIndex.indexer_between_time`, I'm not going to remove the old arguments (`include_start`, `include_end`) for compatibility & add the new `inclusive` argument. Let me know if I should take another course of action.

Reply:
@attack68 Can you take a look at the latest code, please, and let me know should I still modify `DateTimeIndex.indexer_between_time`?

Reply:
No don't do it in this PR. Too complex.
However, once this PR is merged do feel free to do a follow up PR, or post a message in the Issue helping other users to find this and to carry on from your work.

Reply:
Yes, that's the best. I'll do a follow up PR after this.